projects/angular/src/forms/datepicker/date-container.ts

```diff
@@ -175,6 +178,36 @@ export class ClrDateContainer extends ClrAbstractContainer implements AfterViewI
     );
   }
 
+  /**
+   * Return the label for the toggle button.
+   * If there's a selected date, the date is included in the label.
+   */
+  private getToggleButtonLabel(day: DayModel): string {
+    if (day) {
+      const formattedDate = this.dateIOService.toLocaleDisplayFormatString(day.toDate());
+
+      return (
+        this.commonStrings.parse(this.commonStrings.keys.datepickerToggleChangeDateLabel, {
+          SELECTED_DATE: formattedDate,
+        }) || this.commonStrings.keys.datepickerToggle
+      );
+    }
+    return this.commonStrings.keys.datepickerToggleChooseDateLabel || this.commonStrings.keys.datepickerToggle;
+  }
+
+  private listenForDateChanges() {
+    // because date-input.ts initializes the input in ngAfterViewInit,
+    // using a databound attribute to change the button labels results in ExpressionChangedAfterItHasBeenCheckedError.
+    // so instead, update the attribute directly on the element
+    return this.dateNavigationService.selectedDayChange
+      .pipe(startWith(this.dateNavigationService.selectedDay))
+      .subscribe(day => {
+        const label = this.getToggleButtonLabel(day);
+        this.toggleButton.nativeElement.setAttribute('title', label);
+        this.toggleButton.nativeElement.setAttribute('aria-label', label);
+      });
+  }
```

Comment:
I am a bit concerned here as we are accessing DOM API without platform/environment checking. Usually, these kinds of dom changes should be done through [Renderer2](https://angular.io/api/core/Renderer2) API with an additional checking of document being present. You can see an example of how Angular checks whether document is present or not in `datepicker-enabled.service.ts`

Reply:
Ah, good catch. I was initially copying what I saw here in `date-input`, but then totally left out the renderer.setProperty part: https://github.com/vmware/clarity/blob/794ec3e19f3f70a91d6eb15944af6b3f98d28a30/projects/angular/src/forms/datepicker/date-input.ts#L278

Reply:
I'm looking at the file you mentioned, and I'm not sure how it's relevant? It makes sense it's checking for mobile in that case, since it changes how the popup behaves, but I'm not doing that here, just setting an aria label. 

Is switching to using the renderer sufficient? 

```
 const toggleEl = this.toggleButton.nativeElement;
 this.renderer.setAttribute(toggleEl, 'aria-label', label);
```

Reply:
Ah ok, I was thinking SSR might be the reason. I was googling to see if `ngAfterViewInit` gets called during SSR - in react lifecycle events after render don't get called then, so wasn't sure it would be an issue.

Thanks for the explanation